File: Java/1501-1550/1512. Number of Good Pairs.java
// https://leetcode.com/problems/number-of-good-pairs/

class Solution {
    public int numIdenticalPairs(int[] nums) {
        int result = 0;
        HashMap<Integer, Integer> storage = new HashMap<>();

        for(int num : nums){
            result += storage.getOrDefault(num,0);
            storage.put(num, storage.getOrDefault(num, 0)+1);
        }

        return result;
    }
}

File: Python/1501-1550/1512. Number of Good Pairs.py
# https://leetcode.com/problems/number-of-good-pairs/

class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        result = 0
        storage = {}

        for num in nums:
            if num in storage:
                result += storage[num]
                storage[num] += 1
            else:
                storage[num] = 1

        return result
        
